packages/compass-query-bar/src/components/query-option.tsx

```diff
@@ -131,6 +136,17 @@ const QueryOption: React.FunctionComponent<QueryOptionProps> = ({
     [name, onChange]
   );
 
+  const onBlurEditor = useCallback(() => {
+    if (
+      !!editorCurrentValueRef.current &&
+      editorCurrentValueRef.current !== editorInitialValueRef.current &&
+      (editorInitialValueRef.current || editorCurrentValueRef.current !== '{}')
```

Comment:
I'm not sure I understand why we are checking whether or not previous value was empty _or_ that current is not `{}`? Is it supposed to be `!(previous === '' && current === '{}')` (or `previous !== '' || current !== '{}'`) instead?

Reply:
Ah, we caught up, this is similar, just handles `undefined` to